collector/receiver/telemetryapireceiver/factory.go

```diff
@@ -39,14 +40,32 @@ func NewFactory(extensionID string) receiver.Factory {
 				extensionID: extensionID,
 			}
 		},
-		receiver.WithTraces(createTracesReceiver, stability))
+		receiver.WithTraces(createTracesReceiver, stability),
+		receiver.WithLogs(createLogsReceiver, stability))
 }
 
 func createTracesReceiver(ctx context.Context, params receiver.CreateSettings, rConf component.Config, next consumer.Traces) (receiver.Traces, error) {
 	cfg, ok := rConf.(*Config)
 	if !ok {
 		return nil, errConfigNotTelemetryAPI
 	}
+	r := receivers.GetOrAdd(cfg, func() component.Component {
+		return newTelemetryAPIReceiver(cfg, params)
+	})
+	r.Unwrap().(*telemetryAPIReceiver).registerTracesConsumer(next)
```

Comment:
I would want more data protection in this file than I see in the top-level variable for `receivers`, but it looks like the code doesn't read from `receivers`. As I traced my steps through this code, I started thinking we wouldn't get a benefit from the call to `GetOrAdd` receivers. What if we changed that to the following?
```suggestion
	r := newTelemetryAPIReceiver(cfg, params)
	r.Unwrap().(*telemetryAPIReceiver).registerTracesConsumer(next)
```
And if that works we don't have to rely on copying `internal/sharedcomponent`.

Reply:
and actually this is a bit fragile. this means that you cannot use this receiver in lets say, more than one pipeline.

Reply:
I was thinking to reuse the http server. `internal/sharedcomponent` can help to reuse the http server by looking up the receiver using cfg as a key. 
I was assuming we could have configuration for `telemetryapi`

Reply:
I think `sharedcomponent` could play a role so as to reuse http server and to support multiple instances of telemetry api receiver. 

Can you take a look to the updated change and let me know your feedback? Thanks!